vscode-extension/src/editorServer.ts

```diff
@@ -0,0 +1,74 @@
+import * as vscode from "vscode";
+import { getPortPromise } from "portfinder";
+import { EXTENSION_NAME } from "./util";
+import { getPythonPath } from "./utilities/pythonSetupUtils";
+import { ChildProcessWithoutNullStreams, spawn } from "child_process";
+
+/**
+ * Provider for AIConfig editors.
+ *
+ * AIConfig editors are used for `.aiconfig`, `.aiconfig.json` and `.aiconfig.yaml` files.
+ * These files are backed by a JSON schema under the hood.
+ */
+export class EditorServer {
+  private cwd: string;
+
+  public pid: number | null = null;
+  public port: number | null = null;
+  // TODO: Should make this private and expose subscriptions to .stderr, .stdout, and .on
+  public serverProc: ChildProcessWithoutNullStreams | null = null;
+  public url: string | null = null;
+
+  constructor(workingDirectory: string) {
+    this.cwd = workingDirectory;
+  }
+
+  public async start(): Promise<EditorServer> {
+    if (this.serverProc) {
+      throw new Error("Server process already started");
```

Comment:
nit: throw or just `return this`?

Reply:
Oh good point, can just log and return this